tests/page/page-request-continue.spec.ts

```diff
@@ -412,6 +412,27 @@ it('continue should propagate headers to redirects', async ({ page, server, brow
   expect(serverRequest.headers['custom']).toBe('value');
 });
 
+it('redirected requests should report overridden headers', {
+  annotation: { type: 'issue', description: 'https://github.com/microsoft/playwright/issues/31351' }
+}, async ({ page, server, browserName }) => {
+  it.fixme(browserName === 'firefox');
```

Comment:
What's up with Firefox?

Reply:
The headers are not propagated in firefox apparently (other similar tests also disabled). We can tackle that separately.